src/cascadia/WindowsTerminal/AppHost.cpp

```diff
@@ -915,5 +918,82 @@ void AppHost::_HandleSettingsChanged(const winrt::Windows::Foundation::IInspecta
 void AppHost::_IsQuakeWindowChanged(const winrt::Windows::Foundation::IInspectable&,
                                     const winrt::Windows::Foundation::IInspectable&)
 {
+    if (_window->IsQuakeWindow() && !_logic.IsQuakeWindow())
+    {
+        // If we're exiting quake mode, we should make our
+        // tray icon disappear.
+        if (_trayIconData)
+        {
+            Shell_NotifyIcon(NIM_DELETE, &_trayIconData.value());
+            _trayIconData.reset();
+        }
+    }
+
     _window->IsQuakeWindow(_logic.IsQuakeWindow());
 }
+
+void AppHost::_HandleTrayIconPressed()
+{
+    // Currently scoping "minimize to tray" to only
+    // the quake window.
+    if (_logic.IsQuakeWindow())
+    {
+        _window->SummonWindow({});
+    }
+}
+
+void AppHost::_HandleWindowHidden()
+{
+    if (_logic.IsQuakeWindow())
+    {
+        _UpdateTrayIcon();
+    }
+}
+
+// Method Description:
+// - Creates and adds an icon to the notification tray.
+// Updates the current tray icon with the current window's
+// HWND if an icon already exists.
+// Arguments:
+// - <unused>
+// Return Value:
+// - <none>
+void AppHost::_UpdateTrayIcon()
+{
+    if (!_trayIconData)
+    {
+        NOTIFYICONDATA nid{};
+
+        // This HWND will receive the callbacks sent by the tray icon.
+        nid.hWnd = _window->GetHandle();
+
+        // App-defined identifier of the icon. The HWND and ID are used
+        // to identify which icon to operate on when calling Shell_NotifyIcon.
+        // Multiple icons can be associated with one HWND, but here we're only
+        // going to be showing one so the ID doesn't really matter.
+        nid.uID = 1;
+
+        nid.uCallbackMessage = WM_APP + 1;
+
+        nid.hIcon = static_cast<HICON>(GetActiveAppIconHandle(ICON_SMALL));
+        StringCchCopy(nid.szTip, ARRAYSIZE(nid.szTip), L"Windows Terminal");
+        nid.uFlags = NIF_MESSAGE | NIF_SHOWTIP | NIF_TIP | NIF_ICON;
+        Shell_NotifyIcon(NIM_ADD, &nid);

```

Comment:
Should we be `LOG_IF_WIN32_BOOL_FALSE` / `RETURN_IF_WIN32_BOOL_FALSE`-ing these `Shell_NotifyIcon` calls? It doesn't look like they `SetLastError`, so probs not...

Reply:
It's probably safe to let these fall over and ignore them